### PR TITLE
Add Nuget support

### PR DESCRIPTION
**Pull request by RobiNino**

### CODE REVIEW

File: src/main/utils/nugetUtils.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        } catch (error) {
+            logManager.logError(error, !quickScan);
+            logManager.logMessage('Failed building tree for solution "' + slnFilePath + '", due to the above error. Skipping to next solution... ', 'INFO');
+            return null;
+        }
+
+        if (!nugetList.projects) {
+            logManager.logError(new Error('No projects found for solution "' + slnFilePath + '".'), !quickScan);
+            logManager.logMessage(
+                'Possible cause: The solution needs to be restored. Restore it by running "nuget restore ' + slnFilePath + '.',
```
Comment (yahavi) [inline src/main/utils/nugetUtils.ts:83]:
```suggestion
                'Possible cause: The solution needs to be restored. Restore it by running "nuget restore" from "' + slnFilePath + '".',

```

**Review by yahavi — CHANGES_REQUESTED** [review]
Please also run `npm run format`

File: src/main/utils/nugetUtils.ts (hunk 1 above)
Comment (RobiNino) [reply src/main/utils/nugetUtils.ts:83]: slnFilePath is a direct absolute path to the solution so the proposed command is correct (nuget restore /path/to/example.sln)

Comment (yahavi) [reply src/main/utils/nugetUtils.ts:83]: OK, but it looks like there is still a missing `"` in the end

Comment (RobiNino) [reply src/main/utils/nugetUtils.ts:83]: Right, thanks.

File: src/test/tests/nugetUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        assert.lengthOf(res[1].children, 1);
+        assert.deepEqual(res[0].children[0].label, res[1].children[0].label);
+        let child: DependenciesTreeNode = res[0].children[0];
+        assert.deepEqual(child.componentId, 'MyLogger:1.0.0');
+        assert.deepEqual(child.label, 'MyLogger');
+        assert.deepEqual(child.description, '1.0.0');
+        assert.deepEqual(child.parent, res[0]);
+    });
+
+    async function runCreateNpmDependenciesTrees(componentsToScan: Collections.Set<ComponentDetails>, parent: DependenciesTreeNode) {
```
Comment (yahavi) [inline src/test/tests/nugetUtils.test.ts:83]:
```suggestion
    async function runCreateNugetDependenciesTrees(componentsToScan: Collections.Set<ComponentDetails>, parent: DependenciesTreeNode) {
```

File: src/test/tests/nugetUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        for (let expectedSolutionDir of solutionsDirs) {
+            let expectedSolution: vscode.Uri = vscode.Uri.file(path.join(tmpDir.fsPath, expectedSolutionDir, expectedSolutionDir + '.sln'));
+            assert.isTrue(solutions.contains(expectedSolution), 'Should contain ' + expectedSolution.fsPath);
+        }
+    });
+
+    /**
+     * Test NugetUtils.createDependenciesTrees.
+     */
+    it('Create npm Dependencies Trees', async () => {
```
Comment (yahavi) [inline src/test/tests/nugetUtils.test.ts:55]:
```suggestion
    it('Create NuGet Dependencies Trees', async () => {
```

**Review by yahavi — CHANGES_REQUESTED** [review]

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**